app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java

```diff
@@ -6,56 +6,154 @@
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.MotionEvent;
 import android.view.View;
 import android.view.WindowManager;
-import android.widget.Button;
 import android.widget.EditText;
+import android.widget.ImageButton;
 import android.widget.TextView;
 import android.widget.Toast;
+import java.util.Locale;
+import android.os.Handler;
 
-import org.w3c.dom.Text;
+import com.google.gson.Gson;
+import android.location.Location;
 
-import java.util.Locale;
+/*
+ The code for incrementing/decrementing the fare while holding down
+ the up and down arrows is based on: https://goo.gl/zKpYnX
+ Author: Yar
+ Retrieved on: November 5th, 2016
+  */
 
 public class MakeRequestActivity extends AppCompatActivity {
 
+    private static final int DEFAULT_REQ_CODE = 1;
     final Activity activity = MakeRequestActivity.this;
+    /**
+     * Determines how fast the arrows increment/decrement the estimated fare
+     */
+    final int REP_DEL = 25;
+
     private Location start = null;
     private Location end = null;
     private int fareEstimated = -1;
 
+    private Handler repeatUpdateHandler = new Handler();
+    private boolean autoIncrement = false;
+    private boolean autoDecrement = false;
+
+    class RepeatUpdater implements Runnable {
+        public void run() {
+            if(autoIncrement) {
+                incrementFare(null);
+                repeatUpdateHandler.postDelayed( new RepeatUpdater(), REP_DEL);
+            } else if(autoDecrement) {
+                decrementFare(null);
+                repeatUpdateHandler.postDelayed( new RepeatUpdater(), REP_DEL);
+            }
+        }
+    }
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_make_request);
         setTitle("New Request");
-
+        setButtons();
         activity.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
     }
 
+    // TODO instead of location tuples, allow user to view start-to-end path on a map
+    // from: http://stackoverflow.com/questions/14292398/how-to-pass-data-from-2nd-activity-to-1st-activity-when-pressed-back-android
+    // author: ρяσѕρєя K
+    // retrieved on: November 7th, 2016
+    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        super.onActivityResult(requestCode, resultCode, intent);
+        if(requestCode == DEFAULT_REQ_CODE) {
```

Comment:
What? Why is this a thing? I made a comment about your declaration above... maybe I will find more below...